src/criteriaBuilder.js

```diff
@@ -0,0 +1,69 @@
+import extend from 'extend';
+
+export class CriteriaBuilder {
+
+  updateCriteria() {
+    let blocks = [];
+
+    this.filters.forEach((block, index) => {
+      let filtersParsed = {};
+
+      block.forEach(filter => {
+        if (!this.hasValue(filter.data.value)) {
+          return;
+        }
+
+        filtersParsed = extend(filtersParsed, this.parseField(filter.data.field, this.parseOperator(filter.data)));
+      });
+
+      blocks.push(filtersParsed);
+    });
+
+    this.criteria = (blocks.length > 1) ? {where: {or: blocks}} : {where: blocks[0]};
```

Comment:
Oof. Difficult. We could create criteria builders for backends I suppose. If so, I do vote for _only_ building one for sails. Support for others can be added by those who need it.